src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -29,7 +29,7 @@ class ArrayNode extends BaseNode implements PrototypeNodeInterface
     protected $addIfNotSet = false;
     protected $performDeepMerging = true;
     protected $ignoreExtraKeys = false;
-    protected $removeExtraKeys = true;
+    protected $removeExtraKeys = false;
```

Comment:
When I thought about that earlier, I was of the opinion that if someone was ignoring values before, `setIgnoreExtraKeys` must be called to do that and the `$remove` parameter was added with the default value of `true`, which would keeps BC then. When rethinking it now, someone could inherit from `ArrayNode` and change `$ignoreExtraKeys` via other methods as it is `protected`, although this should be an edge case.

But I think setting the initial value of this property to `true` requires no other changes, as it also throws an exception on extra keys when not ignoring them.